src/main/java/io/confluent/idesidecar/restapi/messageviewer/RecordDeserializer.java

```diff
@@ -162,8 +164,17 @@ private JsonNode handleProtobuf(
     try (var protobufDeserializer = new KafkaProtobufDeserializer<>(sr)) {
       protobufDeserializer.configure(SERDE_CONFIGS, isKey);
       var protobufMessage = (DynamicMessage) protobufDeserializer.deserialize(topicName, bytes);
+
+      // Add the message and its nested types to the type registry
+      // used by the JsonFormat printer.
+      var typeRegistry = JsonFormat.TypeRegistry
+          .newBuilder()
+          .add(protobufMessage.getDescriptorForType())
+          .add(protobufMessage.getDescriptorForType().getNestedTypes())
+          .build();
       var printer = JsonFormat
           .printer()
+          .usingTypeRegistry(typeRegistry)
```

Comment:
If this `JsonFormat.TypeRegistry` instance were not used here, the JSON printing for messages having `Any` values would fail with [`Cannot find type for url: <type url>`](https://github.com/protocolbuffers/protobuf/blob/main/java/util/src/main/java/com/google/protobuf/util/JsonFormat.java#L885).